packages/server/modules/workspaces/tests/integration/invites.graph.spec.ts

```diff
@@ -598,6 +603,178 @@ describe('Workspaces Invites GQL', () => {
+          ),
+          leaveStream(
+            myProjectInviteTargetBasicProject,
+            workspaceMemberWithNoProjectAccess
+          )
+        ])
+      })
+
+      afterEach(async () => {
+        await truncateTables([ServerInvites.name])
+      })
+
+      it("can't invite to workspace project through base project invite resolver", async () => {
+        const res = await gqlHelpers.createDefaultProjectInvite({
+          projectId: myProjectInviteTargetWorkspaceProject.id,
+          input: {
+            userId: otherGuy.id,
+            role: Roles.Stream.Owner
+          }
+        })
+
+        expect(res).to.haveGraphQLErrors('Target project belongs to a workspace')
+        expect(res.data?.projectMutations.invites.create.id).to.not.be.ok
+      })
+
+      it('can invite to non-workspace project through workspace project invite resolver', async () => {
+        const res = await gqlHelpers.createWorkspaceProjectInvite({
+          projectId: myProjectInviteTargetBasicProject.id,
+          inputs: [
+            {
+              userId: otherGuy.id,
+              role: Roles.Stream.Owner,
+              workspaceRole: Roles.Workspace.Admin // should be ignored
+            }
+          ]
+        })
+
+        expect(res).to.not.haveGraphQLErrors()
+        expect(res.data?.projectMutations.invites.createForWorkspace.id).to.be.ok
+      })
+
+      it("can't indirectly invite to workspace if not workspace admin", async () => {
+        const res = await gqlHelpers.createWorkspaceProjectInvite(
+          {
+            projectId: myProjectInviteTargetWorkspaceProject.id,
+            inputs: [
+              {
+                userId: otherGuy.id,
+                role: Roles.Stream.Owner
+              }
+            ]
+          },
+          {
+            context: {
+              userId: myWorkspaceFriend.id
+            }
+          }
+        )
+
+        expect(res).to.haveGraphQLErrors(
+          "Inviter doesn't have admin access to the workspace"
+        )
+        expect(res.data?.projectMutations.invites.createForWorkspace.id).to.not.be.ok
+      })
+
+      it('can invite to workspace project even if not workspace admin, if target already belongs to workspace', async () => {
+        const res = await gqlHelpers.createWorkspaceProjectInvite(
+          {
+            projectId: myProjectInviteTargetWorkspaceProject.id,
+            inputs: [
+              {
+                userId: workspaceMemberWithNoProjectAccess.id,
+                role: Roles.Stream.Owner
+              }
+            ]
+          },
+          {
+            context: {
+              userId: myWorkspaceFriend.id
+            }
+          }
+        )
+
+        expect(res).to.not.haveGraphQLErrors()
+        expect(res.data?.projectMutations.invites.createForWorkspace.id).to.be.ok
+      })
+
+      it("can't invite invalid domain email to domain protected workspace project", async () => {
```

Comment:
Validates that workspace project invites also follow workspace domain policy restrictions